static/js/apps/eval_retrieval_generation/sxs/data_store.tsx

```diff
@@ -43,6 +44,38 @@ function getRatingRef(
   );
 }
 
+/**
+ * Saves a rater's input to Firestore only if it is different from any existing
+ * stored rating. Overwrites any previous data.
+ */
+export async function saveRatingIfChanged(
```

Comment:
I don't think this function is necessary and we don't need the extra read because from the component that calls this function: 
- we've already retrieved the stored rating beforehand (can save that in the state or have a bit that says whether rating has changed or something)
- then from the component can call saveRatingToStore only if there is a change

alternatively can just always write to the document since it seems like you're just saving the most recent version anyways so it doesn't matter if it's equivalent or not?